pam/pam.go

```diff
@@ -183,7 +183,9 @@ func pam_sm_authenticate(pamh *C.pam_handle_t, flags, argc C.int, argv **C.char)
 		case StageChallenge:
 			var iaResp *authd.IAResponse
 			var err error
-			// TODO: handle context cancellation to return to previous authentication method selection.
+
+			// Handle signals to return to previous authentication method selection.
+			go cancelIsAuthorizedWatcher(client, sessionID)
```

Comment:
Please do not do this in a goroutine: we need to ensure that for a session, we are in control of our signal orders, so we need to wait for the cancel to returns before going on.

Also, here, you spawn a goroutine uncontrolled, this is not a good practice :)

I would not cancel also if there is no isAuthorized in progress though.

Reply:
I've added it there mainly for testing purposes and to show you moreless how I envisioned it being used, as I knew this wouldn't be our final UI with the user (definitely should have commented on it, sorry)